assets/js/util.js

```diff
@@ -12,6 +12,15 @@ var TEXTAREA_AUTO_RESIZE_MINIMUM_WIDTH = 768,
     APY_REQUEST_URL_THRESHOLD_LENGTH = 2000, // maintain 48 characters buffer for generated parameters
     DEFAULT_DEBOUNCE_DELAY = 100;
 
+var INSTALLATION_NOTIFICATION_REQUESTS_BUFFER_LENGTH = 10,
+    INSTALLATION_NOTIFICATION_INDIVIDUAL_DURATION_THRESHOLD = 4000,
+    INSTALLATION_NOTIFICATION_CUMULATIVE_DURATION_THRESHOLD = 3000,
+    INSTALLATION_NOTIFICATION_DURATION = 10000;
+
+var apyRequestTimeout, apyRequestStartTime, installationNotificationShown = false,
+    lastNAPyRequestDurations = [], cumulativeAPyRequestDuration = 0;
```

Comment:
Do we need `cumulativeAPyRequestDuration`? Seems like we could just compute it on demand with a `reduce`?